services/productService.js

```diff
@@ -57,8 +57,38 @@ const sellProduct = async (req, res, next) => {
   }
 };
 
+const purchaseProduct = async (req, res, next) => {
+  let product;
+  try {
+    const { id } = req.params;
+    const { body } = req;
+    const { user } = res.locals;
+    const quantity = body?.productQty
+      ? Number.parseInt(body?.productQty, 10)
+      : null;
+    if (!quantity) {
+      throw new Error('Missing product quantity');
+    }
+    product = await productModel.editProductQuantity(id, {
```

Comment:
The reason that line is needed is because ...

      shelfCount: product.shelfCount,
      inventoryCount: product.inventoryCount + quantity,

These two lines use "product" from that line to get "product.shelfCount" and "product.inventoryCount"